### ISSUE TITLE
Set Random Seed

### ISSUE THREAD

**Issue by EthanDorr** [fmcnallyi3/icetop-cnn#12]
Currently, the training/assessment split is made consistent by setting the random seed for NumPy before taking the split. What we want is for the entire training/assessment process to be reproducible given the same inputs. This is so we can objectively measure whether any changes to the model/training process improve or hinder the performance of a model.

To remedy this, we need to set the random seeds for Python, NumPy, and TensorFlow. There is a function for each to do this. Setting the random seed should be done as early in the script as makes sense, i.e., before any randomization takes place.

It is important that we do not remove the existing random seeding from the training/assessment split, so as to not break existing functionality.

**Comment by tfutrell** [comment]
I made some progress on this. I have a new key 'seed' under PREP in config.py. get_training_assessment_cut in utils.py now checks to see if the prep parameter has a 'seed' key, and uses that as its seed: otherwise, it uses 1148, to keep its original functionality. trainer.py and model.py also check to see if cg.PREP has the key, and use tf.keras.utils.set_random_seed() to set the seed for Python, NumPy, and TensorFlow: otherwise, it uses a random seed just like it currently does.

I have trained six energy models to test this, the first two on seed 1024, the third and fourth on 8192, the fifth on 2048, and the sixth with the seed commented out in the config (randomized). Results have been as expected, except for the fifth one. The model trained completely, and, based on the loss graph, it trained differently than the other models, but I cannot evaluate it in the notebook.

**Comment by fmcnallyi3** [comment]
It looks like setting the random seed is working, but did this code ever get pulled upstream? Please clearly state when it was added to the repository.
